test/e2e/specs/editor/various/writing-flow.spec.js

```diff
@@ -0,0 +1,1004 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	writingFlowUtils: async ( { page }, use ) => {
+		await use( new WritingFlowUtils( { page } ) );
+	},
+} );
+
+test.describe( 'Writing Flow', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Should navigate inner blocks with arrow keys', async ( {
+		editor,
+		page,
+		writingFlowUtils,
+	} ) => {
+		// Assertions are made both against the active DOM element and the
+		// editor state, in order to protect against potential disparities.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/18928
+		let activeElementText;
+
+		await writingFlowUtils.addDemoContent();
+
+		// Arrow up into nested context focuses last text input.
+		await page.keyboard.press( 'ArrowUp' );
+		await expect
+			.poll( writingFlowUtils.getActiveBlockName )
+			.toBe( 'core/paragraph' );
+		activeElementText = await page.evaluate(
+			() => document.activeElement.textContent
+		);
+		expect( activeElementText ).toBe( '2nd col' );
```

Comment:
**Nit**: Just a thought, maybe it's better to use `expect.poll` here to auto wait for the assertion.

```js
await expect.poll(
	() => page.evaluate( () => document.activeElement.textContent )
).toBe( '2nd col' );
```

Preferably, IMO, I'd like to split this into two separate assertions: one for the expected `activeElement`, and the other for the expected text.

```js
await expect( theActiveElementLocator ).toBeFocused();
await expect( theActiveElementLocator ).toHaveText( '2nd col' );
```

Reply:
Updated to use first suggestion. I can't use a normal locator for blocks since we have 4 paragraph blocks.

Is there a better method to get current active element?

Reply:
Maybe we can try `page.locator( ':focus' )`? I've not tried it before though.

Reply:
It worked [4b45e2f](https://github.com/WordPress/gutenberg/pull/43637/commits/4b45e2f92e964e744651c21f6881b22a23c594f6) 🎉  I also wasn't aware that I could reuse the locators. That's pretty cool.

I'm referring to this part of the official docs:
> Every time locator is used for some action, up-to-date DOM element is located in the page.
> - https://playwright.dev/docs/locators